ParabolInc/parabol
# fix: do not cancel stripe subscription in case of failed payment

Fix #8166

I changed the following [subscriptions settings](https://dashboard.stripe.com/settings/billing/automatic) on stripe, it should not affect prod, as we currently terminate subscriptions from the code on failed payments anyway:

<img width="998" alt="image" src="https://github.com/ParabolInc/parabol/assets/466991/f77660ac-2ff4-4d42-bba3-2654e5849439">

<img width="1001" alt="image" src="https://github.com/ParabolInc/parabol/assets/466991/bd30685d-c10f-44d0-9f65-c9f8e006756d">



How to test:

Case 1:

- Setup webhook (see #8072)
- Create a new org
- Upgrade using `4000000000000341` card (It will fail later after upgrade)
- Payment failed event should be received on the webhook
- Refresh page, see teams are locked
- Open stripe, see that subscription is canceled
- Upgrade again with working card `4242 4242 4242 4242`
- See that a new `active` subscription was created

Case 2: 

* make sure the webhook is active
* create a new org
* Uncomment: `// trial_end: toEpochSeconds(new Date(Date.now() + 1000 * 10))`, in `createTeamSubscription`
* Upgrade with [4000 0000 0000 0341](https://stripe.com/testing#cards) card (will fail after trial)
* Wait few minutes
* Go to the subscription https://dashboard.stripe.com/test/subscriptions
* Find draft invoice (or wait for it to be appeared)
* Finalize the invoice — edit and click charge (to avoid waiting 1 hour)
* See that teams were locked
* See that the subscription went to past due state but not canceled
* Update card on organization: `4242 4242 4242 4242`
* See all teams unlocked, subscription became `active`, no new subscription created, invoice is paid
